subworkflows/local/amp.nf

```diff
@@ -19,6 +21,35 @@ workflow AMP {
         ch_versions = ch_versions.mix(MACREL_CONTIGS.out.versions)
     }
 
+    if ( !params.amp_skip_hmmsearch ) {
+        if (params.amp_hmmsearch_models) { ch_amp_hmm_models = Channel.fromPath( params.amp_hmmsearch_models, checkIfExists: true ) } else { exit 1, '[nf-core/funscan] error: hmm model files not found for --amp_hmmsearch_models! Please check input.' }
+
+        ch_amp_hmm_models_meta = ch_amp_hmm_models
+            .map {
+                file ->
+                    def meta  = [:]
+                    ext_index       = file.getName().lastIndexOf(".")
+                    ext             = file.getName().substring(ext_index)
+                    meta['id'] = ext == '.gz' ? file.getName() - '.hmm.gz' :  file.getName() - '.hmm'
```

Comment:
https://www.nextflow.io/docs/latest/script.html#check-file-attributes

```suggestion
                    meta['id'] = file.extension == '.gz' ? file.name - '.hmm.gz' :  file.name - '.hmm'
```

Reply:
Ooh not seen that before (or i forgot)! Cool! Thanks!

Reply:
Nope not seen before... I wonder if this is worth posting on the nf-core slack as s reminder, I've seen other people implementing many different ways of stuff like getting extensions (and getting the dodgy AF stackoverflow posts from 2011...)

Reply:
Heh, that's why I suggested introducing some utility classes/functions that provide these functions https://nfcore.slack.com/archives/CJRH30T6V/p1646386708818919 but it was received with mixed feelings.

Reply:
Hmm I see. 

Maybe you just make a PR for a few simple ones, I think often people are more willing to accept stuff if the solution is in front of their nose ;).

At least I think detecting the extension or removing the extension would be nice. But the questoin where to store it would be the question...

Reply:
OK NGL - I just came across a usecase for this in Taxprofiler as well.. but couldn't find the description of those attributes in the NXF docs from my head nor search... had to come back here. No wonder people aaren't so aware of it :grimacing:

Reply:
I'll see if I can put together some kind of utility class.